modules/puppet_proxy_legacy/class_scanner_base.rb

```diff
@@ -25,5 +25,9 @@ def classes(paths)
     def classes_in_environment(an_environment)
       classes(environments_retriever.get(an_environment).paths)
     end
+
+    def class_count(environment)
+      0 # Not implemented
```

Comment:
I wonder if it should eventually lead to a HTTP 501 Not Implemented instead. Then it's up to the UI to display 0 or a more detailed message.

Reply:
Those are for what I consider "legacy" providers, I'm not even sure how many people still use them...

Reply:
I still think it's cleaner to return nil and raise that as a HTTP 501 (or similar).

Reply:
It is cleaner, but it changes the api contract in a way that's not useful in the future for the sake of small number of users (I assume there aren't many people left using puppet 3 or puppet < 4.4) - I don't see us not supporting these new calls in newer versions of providers.